include/boost/geometry/algorithms/detail/relate/follow_helpers.hpp

```diff
@@ -118,28 +110,167 @@ struct for_each_disjoint_geometry_if<OpId, Geometry, Tag, true>
             detected_intersections[index] = true;
         }
 
-        bool found = false;
-
         // O(N)
         // check predicate for each contained geometry without generated turn
-        for ( std::vector<bool>::iterator it = detected_intersections.begin() ;
-              it != detected_intersections.end() ; ++it )
+        for (std::size_t index = 0; index < detected_intersections.size(); ++index)
         {
             // if there were no intersections for this multi_index
-            if ( *it == false )
+            if (detected_intersections[index] == false)
             {
-                found = true;
-                std::size_t const index = std::size_t(std::distance(detected_intersections.begin(), it));
-                bool cont = pred(range::at(geometry, index));
-                if ( !cont )
+                if (! pred(range::at(geometry, index)))
+                {
                     break;
+                }
             }
         }
-        
-        return found;
     }
 };
 
+
+// In case it was needed in the future. Commented-out for now.
```

Comment:
But it's quite long...
Can't we do this otherwise?

Reply:
Removed.